public/js/components/specs/app.spec.js

```diff
@@ -0,0 +1,243 @@
+import { mount, shallowMount } from "@vue/test-utils";
+import App from "../App";
+import { TITLE_EMOJI_REGEX } from "./../../utils/constants.js";
+
+// Mock dashboard data
+const mockDashboardData = {
+  dashboard: {
+    bgColor: "#808900",
+    bgImageRepeat: "true",
+    bgImageUrl: "",
+    blockSize: [250, 200],
+    title: "\u2700 IoT Dashboard",
+    tiles: [
+      {
+        buttonText: "stop",
+        deviceId: "AZ3166",
+        deviceMethod: "stop",
+        id: "2471d5ab-0d73-42a3-ba4f-f694574feb6b",
+        position: [54, 466],
+        size: [0.8, 0.7],
+        title: "MXChip sending",
+        type: "button"
+      },
+      {
+        id: "84de1d0d-d1ae-4daa-9540-179e9dd4155c",
+        position: [50, 730],
+        size: [1, 1],
+        title: "",
+        type: "sticker",
+        url: "https://upload.wikimedia.org/wikipedia/commons/3/3a/Cat03.jpg"
+      }
+    ],
+    editMode: "unlocked"
+  }
+};
+
+// Mock device data
+const mockDeviceList = ["AZ3166", "Tessel2", "Jenn"];
+
+describe("Number card", () => {
+  test("component can mount", () => {
+    const wrapper = shallowMountApp();
+
+    expect(wrapper.isVueInstance()).toBeTruthy();
+  });
+
+  test("if child components mount", () => {
+    const wrapper = shallowMountApp();
+
+    expect(wrapper.find({ name: "dashboard-settings" }).exists()).toBe(true);
+    expect(wrapper.find({ name: "base-card" }).exists()).toBe(true);
+  });
+
+  test("check before component is created that the default data is clean", () => {
```

Comment:
This doesn't seem to be a robust test. When I run `SIMULATING=true npm t` the environment variable is set correctly in node, and available as `process.env["SIMULATING"]`. In the test however, both the `SIMULATING` global and `vm.simulating` evaluate to `false`, hence the test is passing for the wrong reason. 

Unfortunately I don't have a suggestion on how to fix this, but maybe Jest/Vue testing legend @escherina does.

Reply:
Yeah reading back over this I would have to agree. I think in order for this to become more robust, I need to update the boolean value of SIMULATING and then re test the expected outcome.

i.e.

vm.SIMULATING = true;
expect(vm.simulating).toEqual(false);

vm.SIMULATING = true;
expect(vm.simulating).toEqual(true);

Something like that. ^ is untested so I could be completely wrong. I'll look into this when I get some down time today!

Reply:
According to [this helpful issues](https://github.com/vuejs/vue-test-utils/issues/193#issuecomment-438437626), you should be able to just do `process.env = Object.assign(process.env, { SIMULATING: true });` in a `beforeAll()`. I haven't tested this myself.

Reply:
> According to [this helpful issues](https://github.com/vuejs/vue-test-utils/issues/193#issuecomment-438437626), you should be able to just do `process.env = Object.assign(process.env, { SIMULATING: true });` in a `beforeAll()`. I haven't tested this myself.

Ah, thanks for sending this over. I'll test this and take a deeper look when i update the pr. Thanks for the rec @mpaarating! 🎉

Reply:
> > According to [this helpful issues](https://github.com/vuejs/vue-test-utils/issues/193#issuecomment-438437626), you should be able to just do `process.env = Object.assign(process.env, { SIMULATING: true });` in a `beforeAll()`. I haven't tested this myself.
> 
> Ah, thanks for sending this over. I'll test this and take a deeper look when i update the pr. Thanks for the rec @mpaarating! 🎉

So I've played around with this a bit and there's a few ways to solve this issue, but the underlying issue is that in `App.vue` `simulating` is being set to a string of `true` or `false` rather than a boolean.

You can either cast `SIMULATING` in `jest.config.js` to a string version of `true` or `false` based on `process.env.SIMULATING` or you can cast `simulating` in `App.vue` to a boolean based on the string value of `process.env.SIMULATING`. I personally prefer the cast to a string in the config rather than the messiness of keeping it in `App.vue`.

With that said, I have no idea why its treated as a boolean in one place and string in the other.

Reply:
This is resolved with the recent commit.